x/slashing/keeper/unjail.go

```diff
@@ -30,19 +35,25 @@ func (k Keeper) Unjail(ctx sdk.Context, validatorAddr sdk.ValAddress) error {
 
 	consAddr := sdk.ConsAddress(validator.GetConsPubKey().Address())
 
+	// If the validator has a ValidatorSigningInfo object that signals that the
+	// validator was bonded and so we must check that the validator is not tombstoned
+	// and can be unjailed at the current block.
+	//
+	// A validator that is jailed but has no ValidatorSigningInfo object signals
+	// that the validator was never bonded and must've been jailed due to falling
+	// below their minimum self-delegation. The validator can unjail at point
+	// assuming they've now bonded above their minimum self-delegation.
 	info, found := k.GetValidatorSigningInfo(ctx, consAddr)
-	if !found {
-		return types.ErrNoValidatorForAddress
-	}
-
-	// cannot be unjailed if tombstoned
-	if info.Tombstoned {
-		return types.ErrValidatorJailed
-	}
-
-	// cannot be unjailed until out of jail
-	if ctx.BlockHeader().Time.Before(info.JailedUntil) {
-		return types.ErrValidatorJailed
+	if found {
```

Comment:
The core change is here. We now allow a validator to unjail if no `ValidatorSigningInfo` is present. This should only ever be the case under the condition of the reported issue. Essentially, we allow a validator to unjail immediately (once they've met their minimum self-delegation).